src/diffusers/schedulers/scheduling_flow_match_euler_discrete.py

```diff
@@ -185,23 +197,33 @@ def set_timesteps(
             device (`str` or `torch.device`, *optional*):
                 The device to which the timesteps should be moved to. If `None`, the timesteps are not moved.
         """
-
         if self.config.use_dynamic_shifting and mu is None:
             raise ValueError(" you have a pass a value for `mu` when `use_dynamic_shifting` is set to be `True`")
 
         if sigmas is None:
-            self.num_inference_steps = num_inference_steps
             timesteps = np.linspace(
                 self._sigma_to_t(self.sigma_max), self._sigma_to_t(self.sigma_min), num_inference_steps
             )
 
             sigmas = timesteps / self.config.num_train_timesteps
+        else:
+            num_inference_steps = len(sigmas)
```

Comment:
Just to note passing sigmas to `Euler`  vs `FlowMatchEuler` is different
https://github.com/huggingface/diffusers/blob/074e12358bc17e7dbe111ea4f62f05dbae8a49d5/docs/source/en/using-diffusers/scheduler_features.md?plain=1#L133
https://github.com/huggingface/diffusers/blob/074e12358bc17e7dbe111ea4f62f05dbae8a49d5/src/diffusers/schedulers/scheduling_euler_discrete.py#L367
https://github.com/huggingface/diffusers/blob/074e12358bc17e7dbe111ea4f62f05dbae8a49d5/src/diffusers/pipelines/flux/pipeline_flux.py#L702

Reply:
yes, it is a little bit different (even though I think it should be the same); that's why we created it in the pipeline for convenience (and had planned to move the logic to inside scheduler too, but just haven't done that)

The current logic of `set_timesteps` (for sigma is None path) in FlowMatchEuler is implemented based on SD3( it was the first flow-matching model we have), and it is somehow a little bit different from all the flow-matching models following it; I think the main difference is it applied the shift twice, first time here https://github.com/huggingface/diffusers/blob/074e12358bc17e7dbe111ea4f62f05dbae8a49d5/src/diffusers/schedulers/scheduling_flow_match_euler_discrete.py#L82

and then

https://github.com/huggingface/diffusers/blob/074e12358bc17e7dbe111ea4f62f05dbae8a49d5/src/diffusers/schedulers/scheduling_flow_match_euler_discrete.py#L195

Reply:
BTW, it is very clear the flux method is the "standard" way by now

We had planned to move the flux sigmas to the scheduler, too, but since we want to separate the noise scheduler logic way from the scheduler; maybe we don't have to spend time on that for now because it might be the opposite direction :)

Reply:
oh sorry this here is just explain your question here
> why Flux pipelines pass sigmas and why those are calculated differently to the sigmas is None path 

I think i might put it in the wrong thread here, sorry!